### ISSUE TITLE
[Interface] Decoding Speed Enhancement

### ISSUE THREAD

**Issue by windkit** [leo-project/leo_jerasure#9]
Erasure would reconstruct any missing block during decoding, therefore if we only supply K blocks to decode, the process would be similar to re-do the encode again.
(Supply K data blocks has been optimized in leo_jerasure)

For example in a {4,3} coding, if we supply D1, D2, D3, C0 (Erased D0), now D0, C1, C2 are all rebuilt.
While in the use case of decoding the file, only D0, D1, D2, D3 are necessary.

In short, in leo_jerasure:decode, extra works are done to re-encode the code blocks which is not necessary.

A small experiment with editing the code in JErasure to omit code block re-encoding shows 1.5x improvement in decoding throughput (6GB/s -> 10GB/s)

**Comment by windkit** [comment]
Similar changes to Cauchy-RS shows 2x improvement in throughput (5GB/s -> 10GB/s) {10,4,5} 1-failure
